source/extensions/quic_listeners/quiche/platform/quic_thread_impl.cc

```diff
@@ -0,0 +1,24 @@
+// NOLINT(namespace-envoy)
+//
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "extensions/quic_listeners/quiche/platform/quic_thread_impl.h"
+
+#include <string>
+
+#include "exe/platform_impl.h"
+
+Envoy::Thread::ThreadFactory* getThreadFactory() {
+  static Envoy::PlatformImpl* platform_impl = new Envoy::PlatformImpl();
```

Comment:
cc @jmarantz I haven't been tracking the entire convo here, but don't we want to avoid potentially having multiple platform instances in the process? I thought we were considering some type of `init()` function that the QUICHE platform layer could use to then store static references if needed?

Reply:
This is test-only code, right? Can it be moved to test/extensions/quic_listeners/quiche/platform along with all the other code that depends on it?

Then you can just use `Thread::threadFactoryForTest()` which is defined in `test/test_common/thread_factory_for_test.h`

Reply:
Yes, I moved it to /test.

Reply:
> cc @jmarantz I haven't been tracking the entire convo here, but don't we want to avoid potentially having multiple platform instances in the process? I thought we were considering some type of `init()` function that the QUICHE platform layer could use to then store static references if needed?

Where to inject such init() function then? QUICHE has its own self-sufficient tests. The only place I could think of is overload testing::main, but not sure how much plumbing it would require.

Reply:
My thinking after that last discussion on #6658 was that no init() is necessary as Quic only needs singleton access to the ThreadFactory in tests, and we already have a mechanism for that.

Reply:
If it's just tests, we can do what @jmarantz proposes. If it's in production code, I think an initialize function called from main after the platform is ready should be fine?